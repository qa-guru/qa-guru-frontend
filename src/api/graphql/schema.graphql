schema {
  query: Query
  mutation: Mutation
}

directive @javaType(name: String!) on SCALAR

"Mutation root"
type Mutation {
  """
  user section
  """
  createUser(input: UserCreateInput!): UserDto
  updateUser(input: UserUpdateInput!): UserDto
  updateRole(id: ID!, roles: [UserRole]): UserDto
  lockUser(id: ID!): Void
  unlockUser(id: ID!): Void
  changePassword(newPassword: String!, oldPassword: String!): Void
  changePasswordByUserId(userId: ID!, password: String!): Void
  resetPassword(email: String!): Void
  setPassword(token: String!, newPassword: String!): Void
  """
  lecture section
  """
  updateLecture(input: LectureInput!): LectureInfoDto
  deleteLecture(id: ID!): Void
  """
  lecture home work level section
  """
  updateLectureHomeWorkLevel(
    input: LectureHomeWorkLevelInput!
  ): LectureHomeWorkLevelDto
  deleteLectureHomeWorkLevel(id: ID!): Void
  """
  training purchase section
  """
  updateTrainingPurchase(input: TrainingPurchaseInput!): TrainingPurchaseDto
  """
  training tariff
  """
  updateTrainingTariff(input: TrainingTariffInput!): TrainingTariffDto
  deleteTrainingTariff(id: ID!): Void
  """
  training section
  """
  updateTraining(input: TrainingInput!): TrainingDto
  deleteTraining(id: ID!): Void
  """
  training lecture
  """
  updateTrainingLecture(id: ID!, lectureIds: [ID!]): [TrainingLectureDto]
  """
  studentHomeWork section
  """
  sendHomeWorkToCheck(
    lectureId: ID!
    trainingId: ID!
    content: String!
  ): StudentHomeWorkDto
  updateHomeWork(id: ID!, content: String!): StudentHomeWorkDto
  deleteHomeWork(id: ID!): Void
  takeForReview(homeWorkId: ID!): StudentHomeWorkDto
  approved(homeWorkId: ID!): StudentHomeWorkDto
  notApproved(homeWorkId: ID!): StudentHomeWorkDto
  resetState(homeWorkId: ID!): StudentHomeWorkDto
  """
  commentHomeWork section
  """
  sendComment(homeWorkId: ID!, content: String!): CommentHomeWorkDto
  answerComment(parentID: ID!, content: String!): CommentHomeWorkDto
  updateComment(id: ID!, content: String!): CommentHomeWorkDto
  deleteComment(id: ID!): Void
}

"Query root"
type Query {
  """
  user section
  """
  user: UserDto
  userById(id: ID): UserDto
  userRoles: [UserRoleDto]
  users(offset: Int!, limit: Int!, sort: UserSort): UsersDto
  usersRating(offset: Int!, limit: Int!, sort: UserSort): UsersRatingDto
  mentors(offset: Int!, limit: Int!, sort: UserSort): UsersDto
  checkResetPasswordToken(token: String!): Void
  """
  lecture section
  """
  lecture(id: ID): LectureInfoShortDto
  lectureHomeWork(lectureId: ID): [LectureContentHomeWorkDto]
  lectures(offset: Int!, limit: Int!, sort: LectureSort): LecturesDto
  """
  lecture home work level section
  """
  lectureHomeWorkLevel(id: ID!): LectureHomeWorkLevelDto
  lectureHomeWorkLevels: [LectureHomeWorkLevelDto]
  """
  purchase section
  """
  trainingPurchases: [TrainingPurchaseDto]
  trainingPurchasesByUserId(userId: ID!): [TrainingPurchaseDto]
  """
  training tariff
  """
  trainingTariffs(
    offset: Int!
    limit: Int!
    sort: TrainingTariffSort
  ): TrainingTariffsDto
  """
  training section
  """
  training(id: ID!): TrainingDto
  trainingsByMentor(offset: Int!, limit: Int!, sort: TrainingSort): TrainingsDto
  trainings(offset: Int!, limit: Int!, sort: TrainingSort): TrainingsDto
  """
  training lecture
  """
  trainingLectures(id: ID!): [TrainingLectureDto]
  """
  studentHomeWork section
  """
  homeWork(id: ID!): StudentHomeWorkDto
  homeWorkByStudentAndLectureAndTraining(
    studentId: ID!
    lectureId: ID!
    trainingId: ID!
  ): StudentHomeWorkDto
  homeWorkByLectureAndTraining(
    lectureId: ID!
    trainingId: ID!
  ): StudentHomeWorkDto
  homeWorks(
    offset: Int!
    limit: Int!
    sort: StudentHomeWorkSort
    filter: StudentHomeWorkFilter
  ): StudentHomeWorksDto
  homeWorksByLectureId(
    offset: Int!
    limit: Int!
    sort: StudentHomeWorkSort
    lectureId: ID!
  ): StudentHomeWorksDto
  homeWorksByStatus(
    offset: Int!
    limit: Int!
    sort: StudentHomeWorkSort
    status: StudentHomeWorkStatus!
  ): StudentHomeWorksDto
  """
  commentHomeWork section
  """
  commentHomeWorkById(id: ID!): CommentHomeWorksDto
  commentsHomeWorkByHomeWork(
    offset: Int!
    limit: Int!
    sort: CommentHomeWorkSort
    homeWorkId: ID!
  ): CommentHomeWorksDto
  """
  statistics section
  """
  trainingsHomeWorksStatistic(id: ID): [TrainingHomeWorksStatisticDto]
  homeWorksStatistic(id: ID): [HomeWorksStatisticDto]
  """
  rating
  """
  rating: RatingDto
  ratingByUser(id: ID): RatingDto
}

type TrainingPurchaseDto {
  id: ID
  user: UserDto!
  trainingTariff: TrainingTariffDto!
}

input TrainingPurchaseInput {
  id: ID
  userEmail: String
  trainingTariffCode: String
}

input TrainingTariffInput {
  id: ID
  name: String
  code: String
  price: Float
  homeWork: Boolean
  trainingName: String
  description: String
}

type TrainingTariffDto {
  id: ID
  name: String
  code: String
  price: Float
  homeWork: Boolean
  training: TrainingDto
  description: String
}

type TrainingDto {
  id: ID!
  name: String!
  description: String
  content: String
  techStack: TechStack!
  mentors: [UserDto]
  tariffs: [TrainingTariffDto]
}

input TrainingInput {
  id: ID
  name: String
  description: String
  content: String
  techStack: TechStack!
  mentors: [String]
}

input TrainingLectureInput {
  lecture: ID!
  lastLecture: ID
  locking: Boolean
}

type TrainingLectureDto {
  id: ID
  number: Int
  lecture: LectureDto
  lastLecture: LectureDto
  locking: Boolean
}

input LectureInput {
  id: ID
  homeWorkLevelCode: String
  subject: String
  description: [String]
  content: [LectureContentInput]
  contentHomeWork: [LectureContentHomeWorkInput]
  speakers: [String]
}

type LectureInfoDto {
  id: ID
  homeWorkLevel: LectureHomeWorkLevelDto
  subject: String
  description: [String]
  content: [LectureContentDto]
  contentHomeWork: [LectureContentHomeWorkDto]
  speakers: [UserDto]
  creationDate: LocalDateTime
  modificationDate: LocalDateTime
}

type LectureInfoShortDto {
  id: ID
  homeWorkLevel: LectureHomeWorkLevelDto
  subject: String
  description: [String]
  content: [LectureContentDto]
  speakers: [UserDto]
  creationDate: LocalDateTime
  modificationDate: LocalDateTime
}

type LectureDto {
  id: ID
  homeWorkLevel: LectureHomeWorkLevelDto
  subject: String
  description: [String]
  speakers: [UserDto]
  creationDate: LocalDateTime
  modificationDate: LocalDateTime
}

type LectureHomeWorkLevelDto {
  id: ID
  code: String
  description: String
  estimate: Int
}

input LectureHomeWorkLevelInput {
  id: ID
  code: String
  description: String
  estimate: Int
}

type LecturesDto {
  items: [LectureDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type LectureContentDto {
  type: String
  value: String
  url: String
}

type LectureContentHomeWorkDto {
  type: String
  value: String
  url: String
}

input LectureContentInput {
  type: String
  value: String
  url: String
}

input LectureContentHomeWorkInput {
  type: String
  value: String
  url: String
}

type StudentHomeWorkDto {
  id: ID
  lecture: LectureInfoDto
  training: TrainingDto
  status: StudentHomeWorkStatus
  answer: String
  student: UserDto
  mentor: UserDto
  creationDate: LocalDateTime
  startCheckingDate: LocalDateTime
  endCheckingDate: LocalDateTime
}

type StudentHomeWorksDto {
  items: [StudentHomeWorkDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type CommentHomeWorksDto {
  items: [CommentHomeWorkDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type CommentHomeWorkDto {
  id: ID
  homeWork: StudentHomeWorkDto
  creator: UserDto
  content: String
  children: [CommentHomeWorkDto]
  creationDate: LocalDateTime
}

type TrainingsDto {
  items: [TrainingDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type TrainingTariffsDto {
  items: [TrainingTariffDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type ContentFileDto {
  id: ID
  size: Long
  type: String
  name: String
  fileLocation: String
}

input UserCreateInput {
  id: ID
  email: String!
  password: String!
  firstName: String!
  lastName: String!
  middleName: String
  phoneNumber: String
}

input UserUpdateInput {
  id: ID
  email: String!
  firstName: String!
  lastName: String!
  middleName: String
  phoneNumber: String
}

type UserDto {
  id: ID
  email: String
  firstName: String
  lastName: String
  middleName: String
  phoneNumber: String
  rating: RatingUserDto
  avatarLocation: String
  roles: [UserRole]
  locked: Boolean
  creationDate: LocalDateTime
  confirmationDate: LocalDateTime
  updateDate: LocalDateTime
}

type UserRatingDto {
  id: ID
  firstName: String
  lastName: String
  middleName: String
  rating: RatingUserDto
  avatarLocation: String
}

type UserRoleDto {
  name: String
  description: String
}

type UsersDto {
  items: [UserDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type UsersRatingDto {
  items: [UserRatingDto]
  offset: Int
  limit: Int
  totalElements: Long
}

type TrainingHomeWorksStatisticDto {
  training: TrainingDto
  homeworks: [HomeWorksStatisticDto]
}

type HomeWorksStatisticDto {
  status: StudentHomeWorkStatus
  count: Long
}

type RatingUserDto {
  rating: Long
}

type RatingDto {
  rating: Long
  products: [RatingProductsDto]
}

type RatingProductsDto {
  training: TrainingDto
  roles: [RatingProductsByUserRoleDto]
}

type RatingProductsByUserRoleDto {
  role: UserRoleDto
  rating: Long
  types: [RatingProductsByRatingTypeDto]
}

type RatingProductsByRatingTypeDto {
  type: RatingTypeDto
  rating: Long
}

type RatingTypeDto {
  name: String
  description: String
}

input UserSort {
  field: UserSortField
  order: Order
}

enum TechStack {
  JAVA
  PYTHON
}

enum UserRole {
  ADMIN
  STUDENT
  MANAGER
  MASTER
  MENTOR
  LECTOR
}

enum Order {
  ASC
  DESC
}

enum UserSortField {
  LAST_NAME
  PHONE
  EMAIL
  RATING
}

enum StudentHomeWorkStatus {
  NEW
  IN_REVIEW
  APPROVED
  NOT_APPROVED
}

input LectureSort {
  field: LectureSortField
  order: Order
}

input StudentHomeWorkFilter {
  status: StudentHomeWorkStatus
  trainingId: ID
  lectureId: ID
  mentorId: ID
  creationDateFrom: LocalDateTime
  creationDateTo: LocalDateTime
}

input StudentHomeWorkSort {
  field: StudentHomeWorkSortField
  order: Order
}

input CommentHomeWorkSort {
  field: CommentHomeWorkSortField
  order: Order
}

input TrainingSort {
  field: TrainingSortField
  order: Order
}

input TrainingTariffSort {
  field: TrainingTariffField
  order: Order
}

enum LectureSortField {
  SUBJECT
  CREATION_DATE
}

enum StudentHomeWorkSortField {
  MENTOR
  STUDENT
  STATE
  CREATION_DATE
  START_CHECKING_DATE
  END_CHECKING_DATE
}

enum CommentHomeWorkSortField {
  CREATOR
  CREATION_DATE
}

enum TrainingSortField {
  NAME
  CREATION_DATE
}

enum TrainingTariffField {
  NAME
  CODE
  CREATION_DATE
}

scalar BigDecimal @javaType(name: "java.math.BigDecimal")

scalar BigInteger @javaType(name: "java.math.BigInteger")

scalar Date @javaType(name: "java.time.LocalDate")

scalar DateTime @javaType(name: "java.time.OffsetDateTime")

scalar LocalDateTime @javaType(name: "java.time.LocalDateTime")

scalar LocalTime @javaType(name: "java.time.LocalTime")

scalar Long @javaType(name: "java.lang.Long")

scalar Time @javaType(name: "java.time.OffsetTime")

scalar Timestamp @javaType(name: "java.util.Date")

scalar Url @javaType(name: "java.net.URL")

scalar Void @javaType(name: "java.lang.Void")
