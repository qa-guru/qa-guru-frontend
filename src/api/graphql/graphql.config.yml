schema: schema.graphql
extensions:
  endpoints:
    Default GraphQL Endpoint:
      url: ${VITE_APP_ENDPOINT}${VITE_GRAPHQL_URI}
      headers:
        user-agent: JS GraphQL
      introspect: false
