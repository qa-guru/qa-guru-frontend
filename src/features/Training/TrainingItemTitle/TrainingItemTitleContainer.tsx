import React from "react";
import Spinner from "../../../shared/Spinner/Spinner";
import { useTrainingQuery } from "../../../api/graphql/training/training";
import { useParams } from "react-router-dom";
import TrainingItemTitle from "./TrainingItemTitle";
import NoDataErrorMessage from "../../../shared/NoDataErrorMessage";

const TrainingItemTitleContainer: React.FC = () => {
  const { trainingId } = useParams();

  const { data, loading } = useTrainingQuery({
    variables: { id: trainingId! },
  });

  if (loading) return <Spinner />;
  if (!data) return <NoDataErrorMessage />;

  return <TrainingItemTitle data={data} />;
};

export default TrainingItemTitleContainer;
