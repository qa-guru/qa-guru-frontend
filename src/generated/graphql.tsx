import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
const defaultOptions = {} as const;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  BigDecimal: any;
  BigInteger: any;
  Date: any;
  DateTime: any;
  LocalDateTime: any;
  LocalTime: any;
  Long: any;
  Time: any;
  Timestamp: any;
  Url: any;
  Void: any;
};

export type ContentFileDto = {
  __typename?: 'ContentFileDto';
  fileLocation?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  name?: Maybe<Scalars['String']>;
  size?: Maybe<Scalars['Long']>;
  type?: Maybe<Scalars['String']>;
};

export type HomeWorkDto = {
  __typename?: 'HomeWorkDto';
  contentFiles?: Maybe<Array<Maybe<ContentFileDto>>>;
  description?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  subject?: Maybe<Scalars['String']>;
};

export type HomeWorkInput = {
  description?: InputMaybe<Scalars['String']>;
  id?: InputMaybe<Scalars['ID']>;
  subject?: InputMaybe<Scalars['String']>;
};

export type HomeWorkSort = {
  field?: InputMaybe<HomeWorkSortField>;
  order?: InputMaybe<Order>;
};

export enum HomeWorkSortField {
  CreationDate = 'CREATION_DATE',
  Subject = 'SUBJECT'
}

export type HomeWorksDto = {
  __typename?: 'HomeWorksDto';
  items?: Maybe<Array<Maybe<HomeWorkDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

/** Mutation root */
export type Mutation = {
  __typename?: 'Mutation';
  /** purchase section */
  createPurchase?: Maybe<PurchaseDto>;
  /** training section */
  createTraining?: Maybe<TrainingDto>;
  /** user section */
  createUser?: Maybe<UserDto>;
  deleteHomeWork?: Maybe<Scalars['Void']>;
  deleteTask?: Maybe<Scalars['Void']>;
  deleteUser?: Maybe<Scalars['Void']>;
  /** homeWork section */
  updateHomeWork?: Maybe<HomeWorkDto>;
  /** person section */
  updatePerson?: Maybe<PersonDto>;
  updateRole?: Maybe<UserDto>;
  /** task section */
  updateTask?: Maybe<TaskDto>;
};


/** Mutation root */
export type MutationCreatePurchaseArgs = {
  input: PurchaseInput;
};


/** Mutation root */
export type MutationCreateTrainingArgs = {
  input: TrainingInput;
};


/** Mutation root */
export type MutationCreateUserArgs = {
  password: Scalars['String'];
  username: Scalars['String'];
};


/** Mutation root */
export type MutationDeleteHomeWorkArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationDeleteTaskArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationDeleteUserArgs = {
  id: Scalars['ID'];
};


/** Mutation root */
export type MutationUpdateHomeWorkArgs = {
  input: HomeWorkInput;
};


/** Mutation root */
export type MutationUpdatePersonArgs = {
  input: PersonInput;
};


/** Mutation root */
export type MutationUpdateRoleArgs = {
  id: Scalars['ID'];
  roles?: InputMaybe<Array<InputMaybe<UserRole>>>;
};


/** Mutation root */
export type MutationUpdateTaskArgs = {
  input: TaskInput;
};

export enum Order {
  Asc = 'ASC',
  Desc = 'DESC'
}

export type PersonDto = {
  __typename?: 'PersonDto';
  avatarLocation?: Maybe<Scalars['String']>;
  firstName?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  lastName?: Maybe<Scalars['String']>;
  middleName?: Maybe<Scalars['String']>;
  phoneNumber?: Maybe<Scalars['String']>;
};

export type PersonInput = {
  firstName: Scalars['String'];
  lastName: Scalars['String'];
  middleName?: InputMaybe<Scalars['String']>;
  phoneNumber: Scalars['String'];
};

export type PurchaseDto = {
  __typename?: 'PurchaseDto';
  id?: Maybe<Scalars['ID']>;
  training: TrainingDto;
  trainingTariff: TrainingTariffDto;
  user: UserDto;
};

export type PurchaseInput = {
  trainingId?: InputMaybe<Scalars['ID']>;
  trainingTariffId?: InputMaybe<Scalars['ID']>;
  userId?: InputMaybe<Scalars['ID']>;
};

/** Query root */
export type Query = {
  __typename?: 'Query';
  /** homeWork section */
  homeWork?: Maybe<HomeWorkDto>;
  homeWorks?: Maybe<HomeWorksDto>;
  /** person section */
  person?: Maybe<PersonDto>;
  personByUserId?: Maybe<PersonDto>;
  /** task section */
  task?: Maybe<TaskDto>;
  taskHomeWorks?: Maybe<Array<Maybe<HomeWorkDto>>>;
  tasks?: Maybe<TasksDto>;
  /** user section */
  user?: Maybe<UserDto>;
  users?: Maybe<UsersDto>;
};


/** Query root */
export type QueryHomeWorkArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryHomeWorksArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<HomeWorkSort>;
};


/** Query root */
export type QueryPersonByUserIdArgs = {
  id: Scalars['ID'];
};


/** Query root */
export type QueryTaskArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryTaskHomeWorksArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryTasksArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<TaskSort>;
};


/** Query root */
export type QueryUserArgs = {
  id?: InputMaybe<Scalars['ID']>;
};


/** Query root */
export type QueryUsersArgs = {
  page: Scalars['Int'];
  size: Scalars['Int'];
  sort?: InputMaybe<UserSort>;
};

export type TaskDto = {
  __typename?: 'TaskDto';
  contentFiles?: Maybe<Array<Maybe<ContentFileDto>>>;
  description?: Maybe<Scalars['String']>;
  homeWorks?: Maybe<Array<Maybe<HomeWorkDto>>>;
  id?: Maybe<Scalars['ID']>;
  last?: Maybe<Scalars['ID']>;
  locking?: Maybe<Scalars['Boolean']>;
  speakers?: Maybe<Array<Maybe<UserInfoDto>>>;
  subject?: Maybe<Scalars['String']>;
};

export type TaskInput = {
  description?: InputMaybe<Scalars['String']>;
  homeWorks?: InputMaybe<Array<InputMaybe<Scalars['String']>>>;
  id?: InputMaybe<Scalars['ID']>;
  last?: InputMaybe<Scalars['ID']>;
  locking?: InputMaybe<Scalars['Boolean']>;
  speakers?: InputMaybe<Array<InputMaybe<Scalars['String']>>>;
  subject?: InputMaybe<Scalars['String']>;
};

export type TaskSort = {
  field?: InputMaybe<TaskSortField>;
  order?: InputMaybe<Order>;
};

export enum TaskSortField {
  CreationDate = 'CREATION_DATE',
  Subject = 'SUBJECT'
}

export type TasksDto = {
  __typename?: 'TasksDto';
  items?: Maybe<Array<Maybe<TaskDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

export enum TechStack {
  Java = 'JAVA',
  Python = 'PYTHON'
}

export type TrainingDto = {
  __typename?: 'TrainingDto';
  id: Scalars['ID'];
  name: Scalars['String'];
  tariffs?: Maybe<Array<Maybe<TrainingTariffDto>>>;
  techStack: TechStack;
};

export type TrainingInput = {
  name?: InputMaybe<Scalars['String']>;
  tariffs?: InputMaybe<Array<InputMaybe<TrainingTariffInput>>>;
  techStack: TechStack;
};

export type TrainingTariffDto = {
  __typename?: 'TrainingTariffDto';
  code?: Maybe<Scalars['String']>;
  description?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  name?: Maybe<Scalars['String']>;
};

export type TrainingTariffInput = {
  code?: InputMaybe<Scalars['String']>;
  description?: InputMaybe<Scalars['String']>;
  name?: InputMaybe<Scalars['String']>;
};

export type UserDto = {
  __typename?: 'UserDto';
  email?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  roles?: Maybe<Array<Maybe<UserRole>>>;
};

export type UserIdInput = {
  id?: InputMaybe<Scalars['ID']>;
};

export type UserInfoDto = {
  __typename?: 'UserInfoDto';
  email?: Maybe<Scalars['String']>;
  id?: Maybe<Scalars['ID']>;
  person?: Maybe<PersonDto>;
  roles?: Maybe<Array<Maybe<UserRole>>>;
};

export enum UserRole {
  Admin = 'ADMIN',
  Lector = 'LECTOR',
  Manager = 'MANAGER',
  Master = 'MASTER',
  Mentor = 'MENTOR',
  User = 'USER'
}

export type UserSort = {
  field?: InputMaybe<UserSortField>;
  order?: InputMaybe<Order>;
};

export enum UserSortField {
  Email = 'EMAIL',
  LastName = 'LAST_NAME',
  Phone = 'PHONE'
}

export type UsersDto = {
  __typename?: 'UsersDto';
  items?: Maybe<Array<Maybe<UserInfoDto>>>;
  totalElements?: Maybe<Scalars['Long']>;
  totalPages?: Maybe<Scalars['Int']>;
};

export type SignUpMutationVariables = Exact<{
  username: Scalars['String'];
  password: Scalars['String'];
}>;


export type SignUpMutation = { __typename?: 'Mutation', createUser?: { __typename?: 'UserDto', id?: string | null, email?: string | null, roles?: Array<UserRole | null> | null } | null };


export const SignUpDocument = gql`
    mutation SignUp($username: String!, $password: String!) {
  createUser(username: $username, password: $password) {
    id
    email
    roles
  }
}
    `;
export type SignUpMutationFn = Apollo.MutationFunction<SignUpMutation, SignUpMutationVariables>;

/**
 * __useSignUpMutation__
 *
 * To run a mutation, you first call `useSignUpMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useSignUpMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [signUpMutation, { data, loading, error }] = useSignUpMutation({
 *   variables: {
 *      username: // value for 'username'
 *      password: // value for 'password'
 *   },
 * });
 */
export function useSignUpMutation(baseOptions?: Apollo.MutationHookOptions<SignUpMutation, SignUpMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<SignUpMutation, SignUpMutationVariables>(SignUpDocument, options);
      }
export type SignUpMutationHookResult = ReturnType<typeof useSignUpMutation>;
export type SignUpMutationResult = Apollo.MutationResult<SignUpMutation>;
export type SignUpMutationOptions = Apollo.BaseMutationOptions<SignUpMutation, SignUpMutationVariables>;