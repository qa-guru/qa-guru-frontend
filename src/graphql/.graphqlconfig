{
  "name": "Untitled GraphQL Schema",
  "schemaPath": "schema.graphql",
  "extensions": {
    "endpoints": {
      "Default GraphQL Endpoint": {
        "url": "http://5.161.103.109:8080/graphql",
        "headers": {
          "user-agent": "JS GraphQL"
        },
        "introspect": false
      }
    }
  }
}