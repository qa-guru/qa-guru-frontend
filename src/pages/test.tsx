import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
