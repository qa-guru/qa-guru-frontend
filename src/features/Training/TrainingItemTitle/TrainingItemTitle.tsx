import React from "react";
import { Typography } from "@mui/material";
import { TrainingQuery } from "../../../generated/graphql";

interface ITrainingItemTitle {
  data: TrainingQuery;
}

const TrainingItemTitle: React.FC<ITrainingItemTitle> = ({ data }) => {
  const { training } = data;

  return (
    <Typography mb="20px" variant="h2">
      {training?.name}
    </Typography>
  );
};

export default TrainingItemTitle;
