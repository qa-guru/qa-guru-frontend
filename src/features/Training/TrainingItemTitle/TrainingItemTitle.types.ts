import { TrainingQuery } from "../../../generated/graphql";

export interface ITrainingItemTitle {
  data: TrainingQuery;
}
