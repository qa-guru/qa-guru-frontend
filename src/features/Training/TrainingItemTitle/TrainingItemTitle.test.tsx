import { MemoryRouter } from "react-router-dom";
import { render } from "../../../test/utilsTest";
import TrainingItemTitle from "./TrainingItemTitle";
import { TechStack } from "../../../generated/graphql";

const mockData: any = {
  __typename: "Query",
  training: {
    __typename: "TrainingDto",
    id: "string",
    name: "string",
    content: "string",
    techStack: TechStack,
    tariffs: [
      {
        __typename: "TrainingTariffDto",
        id: "string",
        name: "string",
        code: "string",
        price: "number",
        homeWork: true,
        description: "string",
      },
    ],
    mentors: [
      {
        __typename: "UserDto",
        id: "string",
        firstName: "string",
        lastName: "string",
        middleName: "string",
      },
    ],
  },
};

describe("TrainingItemTitle", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <TrainingItemTitle data={mockData!} />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
