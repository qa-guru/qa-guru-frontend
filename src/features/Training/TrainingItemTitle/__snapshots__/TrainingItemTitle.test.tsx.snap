// Vitest Snapshot v1

exports[`TrainingItemTitle > the component is render 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root MuiTypography-h2 css-13ddgwt-MuiTypography-root"
  >
    string
  </h2>
</DocumentFragment>
`;
